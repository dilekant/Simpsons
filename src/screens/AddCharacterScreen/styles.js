import {StyleSheet} from 'react-native';
import {responsiveScreenHeight} from 'react-native-responsive-dimensions';

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  addCharacterContainer: {
    flex: 1,
    paddingHorizontal: 10,
  },
  inputText: {
    marginTop: 15,
    marginBottom: 5,
    color: '#000000',
    fontSize: responsiveScreenHeight(2),
  },
  input: {},
  inputAbout: {
    height: responsiveScreenHeight(13),
    textAlignVertical: 'top',
  },
  button: {
    marginTop: 15,
  },
  errorText: {
    marginTop: 15,
    color: 'red',
    fontSize: responsiveScreenHeight(2.2),
    textAlign: 'center',
  },
});

export default styles;
