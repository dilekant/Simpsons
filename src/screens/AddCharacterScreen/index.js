import React, {useContext, useEffect, useState} from 'react';
import {View, Text} from 'react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import Input from '../../components/Input';
import Button from '../../components/Button';
import {RefreshContext} from '../../contexts/RefreshContext';
import styles from './styles';

const AddCharacterScreen = ({navigation}) => {
  const {shouldRefresh} = useContext(RefreshContext);
  const [data, setData] = useState([]);
  const [name, setName] = useState('');
  const [job, setJob] = useState('');
  const [about, setAbout] = useState('');
  const [image, setImage] = useState('');
  const [error, setError] = useState('');

  useEffect(() => {
    getSimpsons();
  }, []);

  const getSimpsons = async () => {
    try {
      const value = await AsyncStorage.getItem('SIMPSONS');
      if (value !== null) {
        setData(JSON.parse(value));
      }
    } catch (e) {
      // error reading value
    }
  };

  const setSimpsons = async response => {
    console.log('response setle');
    try {
      const jsonValue = JSON.stringify(response);
      await AsyncStorage.setItem('SIMPSONS', jsonValue);
      shouldRefresh(true);
      navigation.pop();
    } catch (e) {
      // saving error
    }
  };

  const addPress = () => {
    setError('');
    if(name === '' || job === '' || about === '' || image === '') {
      setError('Please fill in the blanks');
    } else {
      const newCharacter = {
        about: about,
        avatar: image,
        job: job,
        name: name,
      };
      data.push(newCharacter);
      setSimpsons(data);
    }
  };

  return (
    <View style={styles.container}>
      <View style={styles.addCharacterContainer}>
        <Text style={styles.inputText}>Name Surname: </Text>
        <Input value={name} onChangeText={setName} style={styles.input} />
        <Text style={styles.inputText}>Job Title: </Text>
        <Input value={job} onChangeText={setJob} style={styles.input} />
        <Text style={styles.inputText}>About Him/Her: </Text>
        <Input
          value={about}
          onChangeText={setAbout}
          style={styles.inputAbout}
          multiline
        />
        <Text style={styles.inputText}>Image Link: </Text>
        <Input value={image} onChangeText={setImage} style={styles.input} />
        {error === '' ? null : <Text style={styles.errorText}>{error}</Text>}
        <Button
          title={'Add Character'}
          style={styles.button}
          onPress={() => addPress()}
        />
      </View>
    </View>
  );
};

export default AddCharacterScreen;
