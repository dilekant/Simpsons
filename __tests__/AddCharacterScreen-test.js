import React from 'react';
import renderer from 'react-test-renderer';
import AddCharacterScreen from '../src/screens/AddCharacterScreen';

test('renders correctly', () => {
  const tree = renderer.create(<AddCharacterScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});