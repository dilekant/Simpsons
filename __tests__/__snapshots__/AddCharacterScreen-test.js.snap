// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 26.68,
          "marginBottom": 5,
          "marginTop": 15,
        }
      }
    >
      Name Surname: 
    </Text>
    <TextInput
      onBlur={null}
      onChangeText={[Function]}
      onFocus={null}
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 6.67,
            "borderWidth": 0.3,
            "color": "#000000",
            "fontSize": 26.68,
            "height": 80.03999999999999,
            "paddingHorizontal": 10,
            "width": "100%",
          },
          Object {},
        ]
      }
      value=""
    />
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 26.68,
          "marginBottom": 5,
          "marginTop": 15,
        }
      }
    >
      Job Title: 
    </Text>
    <TextInput
      onBlur={null}
      onChangeText={[Function]}
      onFocus={null}
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 6.67,
            "borderWidth": 0.3,
            "color": "#000000",
            "fontSize": 26.68,
            "height": 80.03999999999999,
            "paddingHorizontal": 10,
            "width": "100%",
          },
          Object {},
        ]
      }
      value=""
    />
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 26.68,
          "marginBottom": 5,
          "marginTop": 15,
        }
      }
    >
      About Him/Her: 
    </Text>
    <TextInput
      multiline={true}
      onBlur={null}
      onChangeText={[Function]}
      onFocus={null}
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 6.67,
            "borderWidth": 0.3,
            "color": "#000000",
            "fontSize": 26.68,
            "height": 80.03999999999999,
            "paddingHorizontal": 10,
            "width": "100%",
          },
          Object {
            "height": 173.42000000000002,
            "textAlignVertical": "top",
          },
        ]
      }
      value=""
    />
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 26.68,
          "marginBottom": 5,
          "marginTop": 15,
        }
      }
    >
      Image Link: 
    </Text>
    <TextInput
      onBlur={null}
      onChangeText={[Function]}
      onFocus={null}
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 6.67,
            "borderWidth": 0.3,
            "color": "#000000",
            "fontSize": 26.68,
            "height": 80.03999999999999,
            "paddingHorizontal": 10,
            "width": "100%",
          },
          Object {},
        ]
      }
      value=""
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#3084E0",
          "borderRadius": 6.67,
          "height": 80.03999999999999,
          "justifyContent": "center",
          "marginTop": 15,
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 26.68,
          }
        }
      >
        Add Character
      </Text>
    </View>
  </View>
</View>
`;
